diff --git a/main.c b/main.c
index 748d12f..bab87fd 100644
--- a/main.c
+++ b/main.c
@@ -109,7 +109,7 @@ static uint8_t MAIN_InINT;
 
 SBEMU_EXTFUNS MAIN_SbemuExtFun;
 
-static void MAIN_Interrupt();
+void MAIN_Interrupt();
 static void MAIN_InterruptPM();
 static void MAIN_InterruptRM();
 
@@ -910,7 +910,7 @@ int main(int argc, char* argv[])
         printf("Please try use /i5 or /i7 switch, or disable some onboard devices in the BIOS settings to release IRQs.\n");
         return 1;
     }
-    pcibios_enable_interrupt(aui.card_pci_dev);
+    if(aui.card_pci_dev) pcibios_enable_interrupt(aui.card_pci_dev);
 
     printf("Real mode support: ");
     MAIN_Print_Enabled_Newline(enableRM);
@@ -1275,7 +1275,12 @@ static void MAIN_InterruptRM()
     MAIN_InINT &= ~MAIN_ININT_RM;
 }
 
-static void MAIN_Interrupt()
+void MAIN_CovoxPump(void) // called from the Covox raw IRQ0 ISR
+{
+    HDPMIPT_GetInterrupContext(&MAIN_IntContext);
+    MAIN_Interrupt();
+}
+void MAIN_Interrupt()
 {
     if(!(aui.card_infobits&AUINFOS_CARDINFOBIT_PLAYING))
         return;
diff --git a/makefile b/makefile
index 7c2cc8e..6be15b8 100644
--- a/makefile
+++ b/makefile
@@ -5,7 +5,7 @@ DEBUG ?= 0
 
 VERSION ?= $(shell git describe --tags)
 
-INCLUDES := -I./mpxplay -I./sbemu -I./drivers/include
+INCLUDES := -I. -I./mpxplay -I./sbemu -I./drivers/include
 DEFINES := -D__DOS__ -DSBEMU -DDEBUG=$(DEBUG) -DMAIN_SBEMU_VER=\"$(VERSION)\"
 CFLAGS := -fcommon -march=i386 -ffast-math -O2 -flto $(INCLUDES) $(DEFINES)
 LDFLAGS := -lstdc++ -lm -Wno-attributes
@@ -44,6 +44,7 @@ CARDS_SRC := mpxplay/au_cards/ac97_def.c \
 	     mpxplay/au_cards/sc_sbliv.c \
 	     mpxplay/au_cards/sc_via82.c \
 	     mpxplay/au_cards/sc_null.c \
+	     mpxplay/au_cards/sc_covox.c \
 	     mpxplay/au_cards/sc_ymf.c \
 
 CTXFI_SRC := drivers/ctxfi/ctsrc.c \
diff --git a/mpxplay/au_cards/au_cards.c b/mpxplay/au_cards/au_cards.c
index f1043f3..045e175 100644
--- a/mpxplay/au_cards/au_cards.c
+++ b/mpxplay/au_cards/au_cards.c
@@ -88,6 +88,7 @@ extern one_sndcard_info WINDSOUND_sndcard_info;
 extern one_sndcard_info WINWAVOUT_sndcard_info;
 #endif
 
+extern one_sndcard_info COVOX_sndcard_info;
 extern one_sndcard_info NON_sndcard_info;
 #ifndef SBEMU
 extern one_sndcard_info WAV_sndcard_info;
@@ -96,6 +97,9 @@ extern one_sndcard_info NUL_sndcard_info;
 #endif
 
 static one_sndcard_info *all_sndcard_info[]={
+#ifdef AU_CARDS_LINK_COVOX
+ &COVOX_sndcard_info,
+#endif
 #ifdef AU_CARDS_LINK_YMF
  &YMFSB_sndcard_info,
 #endif
diff --git a/mpxplay/au_cards/au_cards.h b/mpxplay/au_cards/au_cards.h
index c616674..e5f997e 100644
--- a/mpxplay/au_cards/au_cards.h
+++ b/mpxplay/au_cards/au_cards.h
@@ -156,6 +156,7 @@ typedef struct au_cardconfig_v0154_s{
   #define AU_CARDS_LINK_WSS     1
   #define AU_CARDS_LINK_GUS     1
   #define AU_CARDS_LINK_SB      1
+  #define AU_CARDS_LINK_COVOX   1
   //#define AU_CARDS_LINK_MIDAS  1
  #endif
  #ifdef AU_CARDS_LINK_PCI
